ROS/controls/src/moteus_wrapper.py

```diff
@@ -0,0 +1,213 @@
+            queue_size=constants.CONTROLS_PKG_SUBSCRIBER_QUEUE_SIZE
+        )
+
+        self.current_orientation_publisher = rospy.Publisher(
+            constants.CONTROLS_PKG_CURRENT_ORIENTATION_TOPIC,
+            CurrentMotorOrientation,
+            queue_size=constants.CONTROLS_PKG_PUBLISHER_QUEUE_SIZE
+        )
+
+    # MARK: - Methods -
+
+    def motor_description_for_id(self, hwid: int):
+        """Fetches the CAN bus lane ID corresponding to the
+        motor with ID _id_
+
+        @param hwid: The hardware ID identifying the motor whose
+        descriptor should be fetched
+
+        @return: An identifier naming a distinct CAN bus lane
+        """
+        return self.id_map[hwid]
+
+    # MARK: - ROS Messages -
+
+    def run_motor_orientation_loop(self):
+        """Starts the infinite loop to continuously
+        publish ROS notifications about the current state
+        of the Moteus motors
+
+        This method puts the program in an infinite
+        loop, during each iteration of which the node publishes
+        the current orientation of the Moteus motors
+        """
+        rate = rospy.Rate(self.hw_refresh_rate)
+
+        # Wait until the motors are ready (if they aren't already)
+        self.moteus_instance.waitUntilReady()
+
+        while (not rospy.is_shutdown()):
+            latest_motor_state = self.moteus_instance.getParsedResults()
+            motor_states = []
+
+            # Each 'motor_desc' is a dictionary corresponding
+            # to the state of a motor
+            for motor_desc in latest_motor_state:
+                motor_state = MotorState()
+                motor_state.position = motor_desc['POSITION']
+                motor_state.velocity = motor_desc['VELOCITY']
+                motor_state.torque = motor_desc['TORQUE']
+                motor_states.append(motor_state)
+
+            msg = CurrentMotorOrientation(numMotors=self.num_motors, states=motor_states)
+            self.current_orientation_publisher.publish(msg)
+            rate.sleep()
+
+    def shutdown_moteus(self):
+        """Safely terminates the node
+        and performs all cleanup necessary
+        with the Moteus motors and ROS
+
+        It is important to invoke this method
+        before terminating the program; otherwise the
+        Moteus motors in communication with this class
+        will not be properly stopped and there is a
+        possible chance of hardware damage
+        """
+        self.moteus_instance.closeMoteus()
+        exit()
+
+    def __desired_orientation_callback(self, msg: DesiredMotorOrientation):
+        """A method which serves as the callback function
+        of the ROS subscriber representing this instance
+        as a listener to the 'desired_orientation' topic
+
+        @param msg: A message emitted by the Intel NUC
+        of type 'MotorOrientation' describing how the motors
+        should be re-oriented
+        """
+        if len(self.id_map) != msg.numMotors:
```

Comment:
Having error checking is costly in time, remove it and just assume that the message will always be correct.